shared-scripts.sh

```diff
@@ -0,0 +1,506 @@
+# which allows us to share caches with jobs in the same batch
+
+# storeCache <local path> <cache location>
+function storeCache {
+    localPath="$1"
+    alias="$2"
+    s3Path="s3://$CACHE_BUCKET_NAME/$CODEBUILD_SOURCE_VERSION/$alias"
+    echo writing cache to $s3Path
+    # zip contents and upload to s3
+    if ! (cd $localPath && tar cz . | aws s3 cp - $s3Path); then
+        echo Something went wrong storing the cache.
+    fi
+    echo done writing cache
+    cd $CODEBUILD_SRC_DIR
+}
+function storeCacheFile {
+    localFilePath="$1"
+    alias="$2"
+    s3Path="s3://$CACHE_BUCKET_NAME/$CODEBUILD_SOURCE_VERSION/$alias"
+    echo writing cache to $s3Path
+    # zip contents and upload to s3
+    if ! (aws s3 cp $localFilePath $s3Path); then
+        echo Something went wrong storing the cache.
+    fi
+    echo done writing cache
+    cd $CODEBUILD_SRC_DIR
+}
+# loadCache <cache location> <local path>
+function loadCache {
+    alias="$1"
+    localPath="$2"
+    s3Path="s3://$CACHE_BUCKET_NAME/$CODEBUILD_SOURCE_VERSION/$alias"
+    echo loading cache from $s3Path
+    # create directory if it doesn't exist yet
+    mkdir -p $localPath
+    # check if cache exists in s3
+    if ! aws s3 ls $s3Path > /dev/null; then
+        echo "Cache not found."
+        exit 0
+    fi
+    # load cache and unzip it
+    if ! (cd $localPath && aws s3 cp $s3Path - | tar xz); then
+        echo "Something went wrong fetching the cache. Continuing anyway."
+    fi
+    echo done loading cache
+    cd $CODEBUILD_SRC_DIR
+}
+function loadCacheFile {
+    alias="$1"
+    localFilePath="$2"
+    s3Path="s3://$CACHE_BUCKET_NAME/$CODEBUILD_SOURCE_VERSION/$alias"
+    echo loading cache file from $s3Path
+    # check if cache file exists in s3
+    if ! aws s3 ls $s3Path > /dev/null; then
+        echo "Cache file not found."
+        exit 0
+    fi
+    # load cache file
+    if ! (aws s3 cp $s3Path $localFilePath); then
+        echo "Something went wrong fetching the cache file. Continuing anyway."
+    fi
+    echo done loading cache
+    cd $CODEBUILD_SRC_DIR
+}
+function _loadTestAccountCredentials {
+    echo ASSUMING PARENT TEST ACCOUNT credentials
+    session_id=$((1 + $RANDOM % 10000))
+    creds=$(aws sts assume-role --role-arn $TEST_ACCOUNT_ROLE --role-session-name testSession${session_id} --duration-seconds 3600)
+    if [ -z $(echo $creds | jq -c -r '.AssumedRoleUser.Arn') ]; then
+        echo "Unable to assume parent e2e account role."
+        return
+    fi
+    echo "Using account credentials for $(echo $creds | jq -c -r '.AssumedRoleUser.Arn')"
+    export AWS_ACCESS_KEY_ID=$(echo $creds | jq -c -r ".Credentials.AccessKeyId")
+    export AWS_SECRET_ACCESS_KEY=$(echo $creds | jq -c -r ".Credentials.SecretAccessKey")
+    export AWS_SESSION_TOKEN=$(echo $creds | jq -c -r ".Credentials.SessionToken")
+}
+
+function _buildLinux {
+    codebuild-breakpoint
+    echo Linux Build
+    yarn --immutable
```

Comment:
I removed the setShell script here since this is no longer necessary for yarn2 since it uses berry